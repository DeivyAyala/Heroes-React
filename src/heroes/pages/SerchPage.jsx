import React from 'react'
import queryString from 'query-string'


import { HeroCard } from '../components/HeroCard'
import { useForm } from '../../hooks/useForm'
import { useLocation, useNavigate } from 'react-router-dom'
import { getHeroByName } from '../helpers'


export const SerchPage = () => {

  const navigate = useNavigate(); //Obtener la navegación
  const location = useLocation() //obtener la información de la ubicación donde nos encontramos 

  const {q = ''} = queryString.parse(location.search) //encontra el query (opcional)

  const heroes = getHeroByName(q)

  const showSerch = (q.length == 0);
  const showError = (q.length > 0 && heroes.length === 0);



  const {serchText, onInputChange, onResetForm} = useForm({
    serchText: ''  //Tiene que ser igual al name 
  });

  const onSerchSubmit = (e) => {
    e.preventDefault();
    // if(serchText.trim().length <= 2) return;
    navigate(`?q=${serchText}`)
    onResetForm()
  }

  return (
    <>
      <h1>Search</h1>
      <hr/>
      <div className="row">
        <div className="col-5">
          <h4>Busquedas</h4>
          <hr/>
          <form onSubmit={onSerchSubmit}>
            <input
              type='text'
              placeholder='Busqueda de hero'
              className='form-control'
              name='serchText'
              autoComplete='off'
              value={serchText}
              onChange={onInputChange}
            />
            <button className='btn btn-outline-primary mt-2'>
              Buscar
            </button>
          </form>
        </div>
        <div className="col-7">
          <h4>Resultados</h4>
          <hr/>
          {/* {
            (q === '')
            ? <div className='alert alert-primary'> Buscar un Hero </div>
            : (heroes.length === 0) 
            && <div className='alert alert-danger'>Hero no encontrado! - <b>{q}</b></div>

          } */}

          <div className="alert alert-primary animate__animated animate__fadeInLeft" style={{display: showSerch ? '' : 'none' }}>
            Buscar Un hero
          </div>
          <div className='alert alert-danger animate__animated animate__fadeInLeft' style={{display: showError ? '' : 'none' }}>
            Hero no encontrado! - <b>{q}</b></div>
         {
          heroes.map(hero => (
            <HeroCard key={hero.id} {...hero}/>
          ))
         }  
        </div>
      </div>
  
    </>
  )
}
