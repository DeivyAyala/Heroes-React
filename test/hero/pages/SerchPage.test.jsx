import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SerchPage } from "../../../src/heroes/pages/SerchPage"

describe('Pruebas en <SerchPage/>', () => { 
    test('debe de mostarse correctamente con valores por defecto', () => { 
        const { container } = render(
            <MemoryRouter>
                <SerchPage/>
            </MemoryRouter>
        )
        // screen.debug()

        expect(container).toMatchSnapshot()
     })

     test('debe de mostarse a batman y el input con el valor de queryString', () => { 
        // render(
        //     <MemoryRouter initialEntries={['/serch?q=batman']}>
        //         <SerchPage/>
        //     </MemoryRouter>
        // )
    
        // const input = screen.getByRole('textbox');
        // expect(input.value).toBe('batman');
        // screen.debug()
     })
 })