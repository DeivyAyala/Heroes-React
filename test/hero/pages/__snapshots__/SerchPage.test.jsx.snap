// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SerchPage/> debe de mostarse a batman  1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Busquedas
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="serchText"
          placeholder="Busqueda de hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeInLeft"
      >
        Buscar Un hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeInLeft"
        style="display: none;"
      >
        Hero no encontrado! - 
        <b />
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SerchPage/> debe de mostarse a batman y el input  1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Busquedas
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="serchText"
          placeholder="Busqueda de hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeInLeft"
      >
        Buscar Un hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeInLeft"
        style="display: none;"
      >
        Hero no encontrado! - 
        <b />
      </div>
    </div>
  </div>
</div>
